internal/ctxt/merge_test.go

```diff
@@ -0,0 +1,129 @@
+package ctxt_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/gophercloud/gophercloud/internal/ctxt"
+)
+
+func TestMerge(t *testing.T) {
+	t.Run("returns values from both parents", func(t *testing.T) {
```

Comment:
Let's just add a test for values that take precedence, as in https://github.com/gophercloud/gophercloud/pull/2883.